test/util/deterministicuuid/deterministicuuid.go

```diff
@@ -0,0 +1,53 @@
+package deterministicuuid
+
+import (
+	"github.com/gofrs/uuid"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+const (
+	_ uint8 = iota
+	CLUSTERS
+	ASYNCOPERATIONS
+	PORTAL
+	GATEWAY
+)
+
+type gen struct {
+	namespace uint8
+	counter   uint16
+}
+
+func New(namespace uint8) uuid.Generator {
+	return &gen{
+		namespace: namespace,
+	}
+}
+
+func (g *gen) NewV1() (uuid.UUID, error) {
+	panic("unimplemented")
+}
+func (g *gen) NewV3(ns uuid.UUID, name string) uuid.UUID {
+	panic("unimplemented")
+}
+func (g *gen) NewV5(ns uuid.UUID, name string) uuid.UUID {
+	panic("unimplemented")
+}
+func (g *gen) NewV6() (uuid.UUID, error) {
+	panic("unimplemented")
+}
+func (g *gen) NewV7(uuid.Precision) (uuid.UUID, error) {
+	panic("unimplemented")
+}
+
+// NewV4 generates a UUID that increments each call, using a counter to specify
+// the last two bytes and namespaced by the first byte.
+func (g *gen) NewV4() (uuid.UUID, error) {
+	g.counter++
+	// 16 bits of uuid ought to be enough for any test :)
+	return uuid.FromBytesOrNil([]byte{
+		g.namespace, g.namespace, g.namespace, g.namespace, g.namespace, g.namespace, g.namespace, g.namespace, g.namespace, g.namespace, g.namespace, g.namespace, g.namespace, g.namespace, byte(uint8(g.counter >> 8)), byte(uint8(g.counter)),
```

Comment:
wouldn't only one `g.namespace` be sufficient here? I don't really understand why we would need to have the namespace on all the other bytes

Reply:
The idea is that it makes it a bit more obvious that it's not random, and makes the namespacing more prominent. E.g. `0c0c0c0c-0c0c-0c0c-0c0c-0c0c0c0c0001` just _looks_ like it has something up with it :)